### PR TITLE
Bug: support private and public npm packages

### PR DESCRIPTION
**Pull request by estherthetesteramido**
To ensure that our steps can be used for both private and public registries, made the `customRegistry` param input an optional.

This will be used to check if we can audit the packages the pipeline also. Currently Azure Pipelines doesn't support auditing of private packages hosted in Azure.

Conditional to check if we need to authenticate for installation and publishing.

### CODE REVIEW

File: azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml [hunk 1]
```diff
@@ -8,13 +8,14 @@
 parameters:
   versionSpec: ""
   workingDirectory: ""
+  customRegistry: ""
```
Comment (ElvenSpellmaker) [inline azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml:11]: Maybe we should default to `useNpmrc`?

**Review by ElvenSpellmaker — COMMENTED** [review]
As an overall point we _really_ need to standardise across one vairable naming scheme... This PR has some with underscores and some camelcase for example.

Some parameters start with a capital (eurgh) and some don't also... 🤔

File: azDevOps/azure/templates/v2/steps/build-node.yml [hunk 2]
```diff
@@ -54,17 +55,20 @@ steps:
       login_azure: true
 
   # Audit dependencies
-  - ${{ if eq(parameters.build_audit, true) }}:
+  # Important: can only audit public packages in Azure Pipelines
+  - ${{ if ne(parameters.customRegistry, 'useNpmrc') }}:
```
Comment (dnitsch) [inline azDevOps/azure/templates/v2/steps/build-node.yml]: people could use registries other than TFS. 
so maybe let's leave the build_audit in there as well as an agnostic override set specifically in the pipeline

**Review by dnitsch — APPROVED** [review]

File: azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml (hunk 1 above)
Comment (dnitsch) [reply azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml:11]: the default should stay empty 
see updated commit

Comment (ElvenSpellmaker) [reply azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml:11]: Any reason why we shouldn't supply sane defaults?

**Comment by ElvenSpellmaker** [comment]
> 
> 
> As an overall point we _really_ need to standardise across one vairable naming scheme... This PR has some with underscores and some camelcase for example.
> 
> Some parameters start with a capital (eurgh) and some don't also... 🤔

@dnitsch Also this ^

(My vote's on underscores and all lowercase for templates FWIW.)

File: azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml (hunk 1 above)
Comment (dnitsch) [reply azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml:11]: we can as long as the caller specifying a value of `""` will actually count as a truthy value and not falsey hence defaulting to useNpmrc when in fact there is no npmrc file. 
if you know that an empty string is a truthy value in TFS YAML then let's do it

Comment (dnitsch) [reply azDevOps/azure/templates/v2/steps/build-audit-dependencies-node.yml:11]: > As an overall point we _really_ need to standardise across one vairable naming scheme... This PR has some with underscores and some camelcase for example.
> 
> Some parameters start with a capital (eurgh) and some don't also... 🤔

It is also my preference to go with the C style snake casing for variable naming convention - but the first iteration was following the MS approach of pascal casing (as in .net) so we can agree on all new code going forward to be snake_cased

**Pull request merged**